### PR TITLE
Issue 3 amor b

### PR DESCRIPTION
**Pull request by mortenjc**

### CODE REVIEW

File: multiblade/MBCaenBase.cpp [hunk 1]
```diff
@@ -48,27 +47,42 @@ CAENBase::CAENBase(BaseSettings const &settings, struct CAENSettings &LocalMBCAE
   Stats.create("receive.dropped", Counters.FifoPushErrors);
   Stats.create("receive.fifo_seq_errors", Counters.FifoSeqErrors);
 
-  Stats.create("receive.packet_bad_header", Counters.PacketBadDigitizer);
+  Stats.create("headers.packet_bad_header", Counters.PacketBadDigitizer);
+  Stats.create("headers.error_version", Counters.ReadoutsErrorVersion);
+  Stats.create("headers.seq_errors", Counters.ReadoutsSeqErrors);
```
Comment (JennyLouise) [inline multiblade/MBCaenBase.cpp:52]: rename to .HeadersError...

File: multiblade/MBCaenBase.cpp [hunk 2]
```diff
@@ -154,122 +168,198 @@ void CAENBase::processing_thread() {
...
 
-      uint64_t efu_time = 1000000000LU * (uint64_t)time(NULL); // ns since 1970
-      flatbuffer.pulseTime(efu_time);
+    while (true) {
+      if (InputFifo.pop(data_index)) { // There is data in the FIFO - do processing
+        auto datalen = RxRingbuffer.getDataLength(data_index);
+        if (datalen == 0) {
+          Counters.FifoSeqErrors++;
+          continue;
+        }
```
Comment (JennyLouise) [inline multiblade/MBCaenBase.cpp:193]: unindent

File: multiblade/MBCaenBase.cpp [hunk 3]
```diff
@@ -154,122 +168,198 @@ void CAENBase::processing_thread() {
...
+              if (Event.ClusterB.hits.size() == 1) {
+                Counters.EventsWireMultSingle++;
+              }
+
+              XTRACE(EVENT, DEB, "Event Valid\n %s", Event.to_string({}, true).c_str());
+
+              uint16_t x{0};
+              uint16_t y{0};
+
+              if (MBCaen.ModuleSettings.Alignment) {
```
Comment (JennyLouise) [inline multiblade/MBCaenBase.cpp:267]: already checking  MBCaen.ModuleSettings.Alignment

File: multiblade/MBCaenInstrument.cpp [hunk 4]
```diff
@@ -143,67 +133,120 @@ void MBCaenInstrument::FixJumpsAndSort(int cassette, std::vector<Readout> &vec)
...
-      LoadAndProcessReadouts(cassette, temp);
-
+      LoadAndProcessReadouts(DigitiserIndex, temp);
       temp.clear();
       temp.push_back(Readout);
     }
     PrevTime = Time;
   }
-  LoadAndProcessReadouts(cassette, temp);
+  std::sort(temp.begin(), temp.end(), compareByTime);
```
Comment (JennyLouise) [inline multiblade/MBCaenInstrument.cpp:142]: move to start of LoadAndProcessReadouts

**Review by JennyLouise — APPROVED** [review]

File: multiblade/MBCaenBase.cpp (hunk 2 above)
Comment (mortenjc) [reply multiblade/MBCaenBase.cpp:193]: check

File: multiblade/MBCaenBase.cpp (hunk 3 above)
Comment (mortenjc) [reply multiblade/MBCaenBase.cpp:267]: check

File: multiblade/MBCaenInstrument.cpp (hunk 4 above)
Comment (mortenjc) [reply multiblade/MBCaenInstrument.cpp:142]: check

**Pull request merged**